src/openssl.c

```diff
@@ -5134,8 +5134,11 @@ _libssh2_supported_key_sign_algorithms(LIBSSH2_SESSION *session,
     (void)session;
 
 #if LIBSSH2_RSA_SHA2
-    if(key_method_len == 7 &&
-       memcmp(key_method, "ssh-rsa", key_method_len) == 0) {
+    if((key_method_len == 7 &&
+       memcmp(key_method, "ssh-rsa", key_method_len) == 0)
+       || (key_method_len == 28 &&
+       memcmp(key_method, "ssh-rsa-cert-v01@openssh.com", key_method_len) == 0)
```

Comment:
Although there is no official documentation which directly mentions that RSA certs can be upgraded in the same way as regular RSA keys, OpenSSH itself has implemented this capability years ago to force use of RSA-SHA2 signatures with certificate keys as well. Please refer to the below PR for this.
[https://github.com/openssh/openssh-portable/commit/4ba0d54794814ec0de1ec87987d0c3b89379b436](url)

This is my understanding. Now, I know that server-sig-algs are different from normal key auth. All OpenSSH versions above 7.8 accept rsa-sha2-* certificates, even though they're not explicitly enumerated in the server-sig-algs list.For that matter, no RSA cert type keys are enumerated in the server-sig-algs list. Also, we are not just interpreting the existence of rsa-sha2 keys in the server-sig-algs configuration, as it also extends support to certificates. This is the rationale behind OpenSSH mandating RSA-SHA2 signatures only on versions >=7.8 and above knowing that the server supports RSA SHA2 certificates on these versions, and not earlier versions.

In essence, this approach is also driven by the server's capabilities knowing what the remote server supports, rather than solely relying on server-sig-algs.

Reply:
Ok, that commit does show RSA1 certs being upgraded. Looks OK to me.